api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -1982,6 +1990,82 @@ public boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType)
 		return dao.getConceptAttributeCount(conceptAttributeType) > 0;
 	}
 
+	/**
+	 * @see org.openmrs.api.ConceptService#getConceptByReference(String conceptRef
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Concept getConceptByReference(String conceptRef) {
+		if (StringUtil.isBlank(conceptRef)) {
+			return null;
+		}
+		Concept cpt = null;
+		//check if input is a valid Uuid
+		if (UUID_FORMAT.matcher(conceptRef).matches()) {
+			cpt = dao.getConceptByUuid(conceptRef);
+			if (cpt != null) {
+				return cpt;
+			}
+		}
```

Comment:
I would use the same check as is used in the HTML form entry to determine whether or not something is a UUID, namely this:

```java
	/***
	 * Determines if the passed string is in valid uuid format By OpenMRS standards, a uuid must be 36
	 * characters in length and not contain whitespace, but we do not enforce that a uuid be in the
	 * "canonical" form, with alphanumerics seperated by dashes, since the MVP dictionary does not use
	 * this format (We also are being slightly lenient and accepting uuids that are 37 or 38 characters
	 * in length, since the uuid data field is 38 characters long)
	 */
	public static boolean isValidUuidFormat(String uuid) {
		if (uuid.length() < 36 || uuid.length() > 38 || uuid.contains(" ") || uuid.contains(".")) {
			return false;
		}
		
		return true;
	}
```

The reason is not every UUID in OMRS is a "canonical" UUID (in particular, CIEL uses 123456AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA as UUIDs and someone somewhere may still be using old fashioned GUIDs).

Reply:
@ibacher Thanks for bringing this up.  I thought about using 		
```
if (uuid.length() < 36 || uuid.length() > 38 || uuid.contains(" ") || uuid.contains(".")) {
```
but it was considered as an overkill by @samuelmale